internal/app/secretless/handlers/mysql/protocol/protocol.go

```diff
@@ -130,14 +130,101 @@ func UnpackOkResponse(packet []byte) (*OkResponse, error) {
 // HandshakeV10 represents sever's initial handshake packet
 // See https://mariadb.com/kb/en/mariadb/1-connecting-connecting/#initial-handshake-packet
 type HandshakeV10 struct {
-	ProtocolVersion    byte
-	ServerVersion      string
-	ConnectionID       uint32
-	ServerCapabilities uint32
-	AuthPlugin         string
-	Salt               []byte
+	Header                 []byte
+	ProtocolVersion        byte
+	ServerVersion          string
+	ConnectionID           uint32
+	ServerDefaultCollation uint8
+	StatusFlags            uint16
+	ServerCapabilities     uint32
+	AuthPlugin             string
+	Salt                   []byte
 }
 
+// RemoveClientSSL removes Client SSL Capability from Server Handshake
+func (serverHandshake *HandshakeV10) RemoveClientSSL() {
+	serverHandshake.ServerCapabilities &^= ClientSSL
+}
+
+func (serverHandshake *HandshakeV10) Pack() ([]byte, error) {
```

Comment:
I'd like to see an ascii table above here that shows the structure of the packet at the byte level.  That way, I could match up each of the steps of the packing to this "picture" and both better understand and verify the code.  As is, the process of verification of this code would be to find the relevant docs myself on mysql code, and then go through them to find the relevant info about the packet structure, and then do the mapping between that and the code myself.